### PR TITLE
Add support for passing trigger information to the handlers

### PR DESCRIPTION
**Pull request by weshaggard**
PTAL @eerhardt @dagood 

This will add some well known arguments `<trigger-repo>`, etc to get information about what triggered the run. I brought this together yesterday while it was top of mind but I will likely finish it after the current servicing releases.

### CODE REVIEW

File: Maestro/src/Microsoft.DotNet.Maestro.WebApi/Models/SubscriptionsModel.cs [hunk 1]
```diff
@@ -52,7 +52,14 @@ public IEnumerable<ISubscriptionHandler> GetHandlers(ModifiedFileModel modifiedF
             {
                 try
                 {
-                    ISubscriptionHandler subscriptionHandler = resolver.Resolve(subscription);
+                    Dictionary<string, string> triggerSubs = new Dictionary<string, string>()
```
Comment (eerhardt) [inline Maestro/src/Microsoft.DotNet.Maestro.WebApi/Models/SubscriptionsModel.cs]: (minor) What do you think about renaming `triggerSubs` to `parameterSubstitutions`?  I was a little thrown off by this variable name.

**Comment by dagood** [comment]
@MichaelSimons, you might be able to get a lot of use out of this too.

**Comment by eerhardt** [comment]
@weshaggard - is this ready to merge?

**Comment by weshaggard** [comment]
@dotnet/eng-first can someone please deploy this update to Maestro?

**Comment by dagood** [comment]
I don't think that team is kept up to date (yet?). @dotnet/dnceng

**Comment by maririos** [comment]
@dagood how do we deploy this to maestro? is there documentation somewhere? FR could help with this

**Comment by dagood** [comment]
I pointed at the docs in my approval message: https://github.com/dotnet/core-eng/blob/master/Documentation/Maestro/deploy.md

**Comment by maririos** [comment]
Thanks! Issue https://github.com/dotnet/core-eng/issues/1623 created to track the deployment

**Pull request merged**